### PR TITLE
[Hub Generated] Review request for Microsoft.Consumption to add version stable/2019-05-01

### PR DESCRIPTION
**Pull request by santoshsinha100**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/santoshsinha100/azure-rest-api-specs/dev-consumption-Microsoft.Consumption-2019-05-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by santoshsinha100** [comment]
Review request for Microsoft.Consumption to add version stable/2019-05-01.
New API added are...
=>Recommendation Single/shared on billing account scope
=>RIUsage Details  on billing account scope
=>RIUsage Summary on billing account scope

Please review and approve this PR to procced with ARM manifest related changes.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json]:
```suggestion
    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/reservationDetails": {
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json]:
```suggestion
    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/reservationSummaries": {
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesDaily.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesDaily.json]:
```suggestion
            "type": "Microsoft.Consumption/reservationSummaries",
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesDailyWithReservationId.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesDailyWithReservationId.json]:
```suggestion
            "type": "Microsoft.Consumption/reservationSummaries",
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesMonthly.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesMonthly.json]:
```suggestion
            "type": "Microsoft.Consumption/reservationSummaries",
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesMonthlyWithReservationId.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/ReservationSummariesMonthlyWithReservationId.json]:
```suggestion
            "type": "Microsoft.Consumption/reservationSummaries",
```

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json
Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json]: commited

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json
Comment (nschonni) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json]:
```suggestion
    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/reservationSummaries": {
```

Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json]: already fixed

**Comment by santoshsinha100** [comment]
Hi Nick,

Can you mark this item complete. I already addressed all your suggestions. We need this to roll out ASAP.

Thanks & Regards
Santosh

**Comment by nschonni** [comment]
Sorry, I'm not an MS employee, but I think there is instructions over here if you need to escalate https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html

**Comment by santoshsinha100** [comment]
Can some one please approve this PR. This is active since last 4 to 5 days.

**Comment by ms-premp** [comment]
We have preview version for usage details already merged in 
https://github.com/Azure/azure-rest-api-specs/pull/6575 
We are moving this to GA and adding one more API in this GA version. Please review.

**Comment by majastrz** [comment]
>     "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/microsoft.consumption/ReservationRecommendations": {

nit - should be camelCase

---
Refers to: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json:968 in d04e302. [](commit_id = d04e302c85bdffe4cb2c5f966faef6fa1df32921, deletion_comment = False)

**Comment by majastrz** [comment]
>         "billingPeriodId": {

This is potentially an SDK breaking change. (When users update the current SDK to one generated from this Swagger, their code won't compile if they are referencing properties that got removed.) What's the plan to deal with that?

---
Refers to: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/consumption.json:1269 in 71e2dcb. [](commit_id = 71e2dcbd4d0f5e6367ed7a6ec72f77356da1f487, deletion_comment = True)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by santoshsinha100** [comment]
Can some one do this review on priority. Its waiting for an approval

**Comment by majastrz** [comment]
> Can some one do this review on priority. Its waiting for an approval

I'm confused by this. Normally a person from the SDK team is assigned to a PR like this and they finish the review from the SDK side once ARM (me this week) signs off. I'll send an email to a contact I know internally (I don't know her GitHub name to just @ mention here.)

**Comment by majastrz** [comment]
If you reach out to me directly via Teams (alias is same as my github name), I can add you to the mail thread.

**Review by yungezz — COMMENTED** [review]
looks good, some minor issue in samples. @santoshsinha100 could you pls resolve conflict? then I'll go to approve and merge the PR.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/AggregatedCostByManagementGroup.json
Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-05-01/examples/AggregatedCostByManagementGroup.json]: Hi I am unable to resolve the conflict in readme.md file. UI doesn't allow me to resolve.  Looking at the conflict, I will go with "### Tag: package-2019-05" not the preview one. Will it be possible to get this resolved at your end .

**Comment by yungezz** [comment]
@santoshsinha100 pls look at the resolved readme.md file. pls update the tag `package-preview-2019-05` to `package-2019-05` at https://github.com/Azure/azure-rest-api-specs/pull/6620/files#diff-19989b3082cb3c89e63e9903a79a80b6R40 to

**Review by yungezz — CHANGES_REQUESTED** [review]
pls review change in readme.md for conflict resolving, and update tag name per need.

**Review by ms-premp — APPROVED** [review]
Thanks @yungezz. Your ReadMe changes looks good.

**Review by yungezz — APPROVED** [review]

**Pull request merged**
